mdtraj/tests/test_trajectory.py

```diff
@@ -36,25 +36,27 @@
 
 fn = get_fn('traj.h5')
 nat = get_fn('native.pdb')
-fd1, temp1 = tempfile.mkstemp(suffix='.xtc')
-fd2, temp2 = tempfile.mkstemp(suffix='.dcd')
-fd3, temp3 = tempfile.mkstemp(suffix='.binpos')
-fd4, temp4 = tempfile.mkstemp(suffix='.trr')
-fd5, temp5 = tempfile.mkstemp(suffix='.h5')
-fd6, temp6 = tempfile.mkstemp(suffix='.pdb')
-fd7, temp7 = tempfile.mkstemp(suffix='.nc')
-fd8, temp8 = tempfile.mkstemp(suffix='.lh5')
-fd9, temp9 = tempfile.mkstemp(suffix='.lammpstrj')
-fd10, temp10 = tempfile.mkstemp(suffix='.xyz')
-
-for e in [fd1, fd2, fd3, fd4, fd5, fd6, fd7, fd8, fd9, fd10]:
-    os.close(e)
+tmpfns = {}
+for suffix, (fd, temp) in {
+      'xtc' : tempfile.mkstemp(suffix='.xtc'),
+      'dcd' : tempfile.mkstemp(suffix='.dcd'),
+      'binpos' : tempfile.mkstemp(suffix='.binpos'),
+      'trr' : tempfile.mkstemp(suffix='.trr'),
+      'h5' : tempfile.mkstemp(suffix='.h5'),
+      'pdb' : tempfile.mkstemp(suffix='.pdb'),
+      'pdb.gz' : tempfile.mkstemp(suffix='.pdb.gz'),
+      'nc' : tempfile.mkstemp(suffix='.nc'),
+      'lh5' : tempfile.mkstemp(suffix='.lh5'),
+      'lammpstrj' : tempfile.mkstemp(suffix='.lammpstrj'),
+      'xyz' : tempfile.mkstemp(suffix='.xyz')}.iteritems():
+    os.close(fd)
+    tmpfns[suffix] = temp
 
 def teardown_module(module):
     """remove the temporary file created by tests in this file
     this gets automatically called by nose"""
 
-    for e in [temp1, temp2, temp3, temp4, temp5, temp6, temp7, temp8]:
+    for e in tmpfns.itervalues():
```

Comment:
no `itervalues` on py3. `.values` is fine, since this is a trivial amount of memory.